### PR TITLE
Sieve: add support for comparator-i;unicode-casemap (RFC 5051)

### PR DESCRIPTION
**Pull request by ksmurchison**

### CODE REVIEW

File: changes/next/sieve-unicode-casemap [hunk 1]
```diff
@@ -0,0 +1,14 @@
+
+Description:
+
+Adds support for comparator-i;unicode-casemap (RFC 9394) to Sieve
+
```
Comment (dilyanpalauzov) [inline changes/next/sieve-unicode-casemap:5]: As far as I see, RFC 9394 is not about Sieve and not about Unicode.

File: docsrc/imap/rfc-support.rst [hunk 2]
```diff
@@ -390,6 +390,10 @@ The following is an inventory of RFCs supported by Cyrus IMAP.
 
     IMAP URL Scheme, updated by :rfc:`5593`.
 
+:rfc:`5051`
+
+    i;unicode-casemap - Simple Unicode Collation Algorithm
+
```
Comment (dilyanpalauzov) [inline docsrc/imap/rfc-support.rst:396]: If RFC 5051 can be supported in the context of Sieve and in the context of IMAP, I suggest adding a note, that narrows the scope to Sieve.

File: changes/next/sieve-unicode-casemap (hunk 1 above)
Comment (rsto) [reply changes/next/sieve-unicode-casemap:5]: Agreed, that looks like it should be 5051 ?

**Review by rsto — CHANGES_REQUESTED** [review]

File: changes/next/sieve-unicode-casemap (hunk 1 above)
Comment (ksmurchison) [reply changes/next/sieve-unicode-casemap:5]: Fixed

File: docsrc/imap/rfc-support.rst (hunk 2 above)
Comment (ksmurchison) [reply docsrc/imap/rfc-support.rst:396]: Fixed

**Review by rsto — APPROVED** [review]

**Pull request merged**